southpark-search/tests/distributed/terraform.tf

```diff
@@ -0,0 +1,204 @@
+  tags = {
+    Name = "flow"
+  }
+}
+
+data "aws_subnet_ids" "default" {
+  vpc_id = "${aws_default_vpc.default_vpc.id}"
+}
+
+
+resource "aws_ecs_cluster" "southpark_cluster" {
+  name = "southpark_cluster"
+}
+
+resource "aws_iam_role_policy_attachment" "ecsTaskExecutionRole_policy" {
+  role       = "${aws_iam_role.ecsExecutionRole.name}"
+  policy_arn = "arn:aws:iam::aws:policy/service-role/AmazonECSTaskExecutionRolePolicy"
+}
+
+resource "aws_iam_role" "ecsExecutionRole" {
+  name               = "ecsExecutionRole"
+  assume_role_policy = "${data.aws_iam_policy_document.assume_role_policy.json}"
+}
+
+data "aws_iam_policy_document" "assume_role_policy" {
+  statement {
+    effect = "Allow"
+    actions = ["sts:AssumeRole"]
+
+    principals {
+      type        = "Service"
+      identifiers = ["ecs-tasks.amazonaws.com"]
+    }
+  }
+}
+
+#Create task
+resource "aws_ecs_task_definition" "southpark_task" {
+  family                   = "southpark_task" 
+  container_definitions    = <<DEFINITION
+  [
+    {
+      "name": "southpark_task",
+      "image": "${aws_ecr_repository.southpark.repository_url}",
+      "essential": true,
+      "portMappings": [
+        {
+          "containerPort": 45678,
+          "hostPort": 45678
+        }
+      ],
+      "memory": 512,
+      "cpu": 256
+    }
+  ]
+  DEFINITION
+  requires_compatibilities = ["FARGATE"] # Stating that we are using ECS Fargate
+  network_mode             = "awsvpc"    # Using awsvpc as our network mode as this is required for Fargate
+  memory                   = 512         # Specifying the memory our container requires
+  cpu                      = 256         # Specifying the CPU our container requires
+  execution_role_arn       = "${aws_iam_role.ecsExecutionRole.arn}"
+}
+
+
+#create load balancer
+resource "aws_alb" "application_load_balancer" {
+  name               = "southpark-lb-tf" # Naming our load balancer
+  load_balancer_type = "application"
+  subnets            = "${data.aws_subnet_ids.default.ids}" 
+  # Referencing the security group
+  security_groups = ["${aws_security_group.load_balancer_security_group.id}"]
+}
+
+resource "aws_lb_listener" "lsr" {
+  load_balancer_arn = "${aws_alb.application_load_balancer.arn}" # Referencing our load balancer
+  port              = "45678"
+  protocol          = "HTTP"
+  default_action {
+    type             = "forward"
+    target_group_arn = "${aws_lb_target_group.target_group.arn}" # Referencing our tagrte group
+  }
+}
+
+resource "aws_lb_target_group" "target_group" {
+  name        = "target-gp"
+  port        = 45678
+  protocol    = "HTTP"
+  target_type = "ip"
+  deregistration_delay = 90
+  vpc_id      = "${aws_default_vpc.default_vpc.id}" # Referencing the default VPC
+  health_check {
+    healthy_threshold   = "3"
+    interval            = "80"
+    protocol            = "HTTP"
+    timeout             = "60"
+    unhealthy_threshold = "2"
+    matcher             = "200-405"
```

Comment:
Maybe we should switch `matcher` back to `200, 301, 302`? I set this from `200-405` only because I wanted to check if the problem was that it was receiving a 404 error